docs/rules/no-empty-pattern.md

```diff
@@ -0,0 +1,42 @@
+# No empty pattern (no-empty-pattern)
+
```

Comment:
```js
let {foo: {}} = obj;
```

This code may look defining a variable `foo`, but not.
Empty patterns really do nothing, but it's similar to a notation of default value, so it causes a mistake.
So it should be warned by linters.